x-pack/plugins/ml/server/saved_objects/sync.ts

```diff
@@ -241,20 +263,66 @@ export function syncSavedObjectsFactory(
       });
     });
 
+    status.savedObjects['anomaly-detector'].forEach((job) => {
+      const type = 'anomaly-detector';
+      const datafeedId = adJobsById[job.jobId]?.datafeedId;
+      if (!datafeedId) {
+        return;
+      }
+
+      if (
+        (job.checks.datafeedExists === true && job.datafeedId === null) ||
+        (job.checks.datafeedExists === true &&
+          job.datafeedId !== null &&
+          adJobsById[job.jobId] &&
+          adJobsById[job.jobId].datafeedId !== job.datafeedId)
+      ) {
```

Comment:
you need to check for `job.checks.datafeedExists === true` only once
```suggestion
      if (
        (job.checks.datafeedExists === true && (job.datafeedId === null ||
          (job.datafeedId !== null &&
          adJobsById[job.jobId] &&
          adJobsById[job.jobId].datafeedId !== job.datafeedId)
      ) {
```